### PR TITLE
Adding docs for migration config property

### PR DESCRIPTION
**Pull request by m57lyra**
Blended blog post info with other available info for this. The concern here is that users will forget about this and not clean up in the near-term. Then, it will not be documented for them to know it is ok to be removed when they eventually do clean up.

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 1]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
 output data set is not skewed, in order to avoid the overhead of hashing and
 redistributing all the data across the network. This can be specified
 on a per-query basis using the ``redistribute_writes`` session property.
+
+``protocol.v1.alternate-header-name``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Type:** ``string``
+
+The 353-e release of Trino changes the HTTP client protocol headers to start
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:45]: The 351 release

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 2]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
+with ``X-Trino-``. Legacy clients can be temporarily supported by setting this
+configuration property to ``Presto``. The preferred approach to migrating from
+versions earlier than 353 is to update all clients together with the release, or
+immediately afterwards, and then remove usage of this property.
+
+Ensure to use this only as a temporary measure to assist in your migration
+efforts, for example if you continue to operate clusters with an older version
+of Trino.
+
+In addition, use a 350-e version of the relevant client tools for best
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:55]: 350

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 3]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
+compatibility. Release 350-e is the last version with support for releases older
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:56]: 350

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 4]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
 redistributing all the data across the network. This can be specified
 on a per-query basis using the ``redistribute_writes`` session property.
+
+``protocol.v1.alternate-header-name``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Type:** ``string``
+
+The 353-e release of Trino changes the HTTP client protocol headers to start
+with ``X-Trino-``. Legacy clients can be temporarily supported by setting this
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:46]: Legacy clients .. maybe we need to spell this out 

Clients for versions 350 and lower expect the HTTP headers to use X-PRESTO, while newer clients expect X-TRINO. You can support these older clients by setting this property to PRESTO.

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 5]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
+configuration property to ``Presto``. The preferred approach to migrating from
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:47]: New paragraph.. 

The preferred approach..

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 6]
```diff
@@ -36,3 +36,24 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
+versions earlier than 353 is to update all clients together with the release, or
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-general.rst:48]: than 351

File: docs/src/main/sphinx/admin/properties-general.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.rst:55]: I kind of think this whole line 55 and 56 can go. We will have this in the migration guide later.

**Review by mosabua — CHANGES_REQUESTED** [review]
Needs a bunch of fixes for starters.

File: docs/src/main/sphinx/admin/properties-general.rst (hunk 4 above)
Comment (m57lyra) [reply docs/src/main/sphinx/admin/properties-general.rst:46]: Will use case consistent with blog post and migration guide though.

File: docs/src/main/sphinx/release/release-353.md [hunk 7]
```diff
@@ -17,6 +17,7 @@
 * Fix rendering of function references in `EXPLAIN` output. ({issue}`6703`)
 * Fix planning failure when `WITH` clause contains hidden columns. ({issue}`6838`)
 * Prevent client hangs when OAuth2 authentication fails. ({issue}`6659`)
+* Add alternate protocol header name ({issue}`6484`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-353.md:20]: Was this really only fixed in 353?

Comment (m57lyra) [reply docs/src/main/sphinx/release/release-353.md:20]: The tag on the PR says 353

Comment (m57lyra) [reply docs/src/main/sphinx/release/release-353.md:20]: Ahh... 351 was under a ``...``

**Review by mosabua — APPROVED** [review]
Looks good. In the longer run we can link to this from a migration guide. This is a first good step to at least have property.

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 8]
```diff
@@ -36,3 +36,22 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
 redistributing all the data across the network. This can be specified
 on a per-query basis using the ``redistribute_writes`` session property.
+
+``protocol.v1.alternate-header-name``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Type:** ``string``
+
+The 351 release of Trino changes the HTTP client protocol headers to start with
+``X-Trino-``. Clients for versions 350 and lower expect the HTTP headers to use
```
Comment (phd3) [inline docs/src/main/sphinx/admin/properties-general.rst:46]: ``HTTP headers to use`` --> ``HTTP headers to start with``

File: docs/src/main/sphinx/admin/properties-general.rst [hunk 9]
```diff
@@ -36,3 +36,22 @@ across nodes in the cluster. It can be disabled, when it is known that the
...
+``X-Presto-``, while newer clients expect ``X-Trino-``. You can support these
+older clients by setting this property to ``Presto``.
+
+The preferred approach to migrating from versions earlier than 351 is to update
+all clients together with the release, or immediately afterwards, and then
+remove usage of this property.
+
+Ensure to use this only as a temporary measure to assist in your migration
+efforts, for example if you continue to operate clusters with an older version
+of Trino.
```
Comment (phd3) [inline docs/src/main/sphinx/admin/properties-general.rst:56]: I feel the later half of the sentence might be confusing. May be removing this sentence, or just adding `Ensure to use this only as a temporary measure to assist in your migration efforts.` - might be better. I'd leave it to @mosabua.

Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-general.rst:56]: I am fine with cutting it short.

**Comment by m57lyra** [comment]
Replaced by PR#7344

**Pull request closed without merging**